library/std/src/sys/unix/os.rs

```diff
@@ -361,6 +362,14 @@ pub fn current_exe() -> io::Result<PathBuf> {
     }
 }
 
+#[cfg(target_os = "nto")]
+pub fn current_exe() -> io::Result<PathBuf> {
+    let mut e = crate::fs::read("/proc/self/exefile")?;
+    // e ends with a null byte, remove it
+    e.pop();
```

Comment:
@gh-tr : The [documentation](https://www.qnx.com/developers/docs/7.1/#com.qnx.doc.neutrino.prog/topic/process_proc_filesystem.html) says nothing about the content of `/proc/self/exefile` being null-terminated, and I think that it would not be technically required (e.g. reading the whole file always produces the whole string in a safe way). Would it make sense to add a check and only remove the last byte in case it is null (just in case this is changed in the future)?

Reply:
On the other hand, supporting new QNX versions will always mean some manual work in libc and libstd and there are tests that would catch this issue.

Reply:
Yes, that sort of concern is why I mentioned it. I would hesitate to rely on other tests simply implicitly catching such a thing, because `\0` is a _valid UTF-8 character_, so Rust does not say you cannot have the zero byte in your String, and certainly doesn't say so about OsStrings. But it might be awkward if most followup manipulations then produced an incorrect PathBuf, or if trying to use the PathBuf with File operations then did not execute the operation correctly.

Reply:
Addressed with https://github.com/rust-lang/rust/pull/106673/commits/b64793961f7c42fe4a33a5c48656bbe82c01c0a9